packages/@aws-cdk/aws-apigateway/lib/access-log.ts

```diff
@@ -0,0 +1,557 @@
+   */
+  public static contextAuthorizerIntegrationLatency() {
+    return '$context.authorizer.integrationLatency';
+  }
+
+  /**
+   * The integration latency in ms.
+   */
+  public static contextIntegrationLatency() {
+    return '$context.integrationLatency';
+  }
+
+  /**
+   * For Lambda proxy integration, this parameter represents the status code returned from AWS Lambda,
+   * not from the backend Lambda function.
+   */
+  public static contextIntegrationStatus() {
+    return '$context.integrationStatus.';
+  }
+
+  /**
+   * The response latency in ms.
+   */
+  public static contextResponseLatency() {
+    return '$context.responseLatency';
+  }
+
+  /**
+   * The response payload length.
+   */
+  public static contextResponseLength() {
+    return '$context.responseLength';
+  }
+
+  /**
+   * The method response status.
+   */
+  public static contextStatus() {
+    return '$context.status';
+  }
+}
+
+/**
+ * A API Gateway custom access log format.
+ */
+export interface AccessLogFormatProps {
+  /**
+   * A API Gateway custom access log format
+   */
+  readonly format: string;
+}
+
+/**
+ * jsonWithStandardFields property
+ */
+export interface IJsonWithStandardFieldProps {
+  /**
+   * If this flag is enabled, $context.identity.sourceIp will be output to the log
+   */
+  ip: boolean,
+  /**
+   * If this flag is enabled, $context.identity.caller will be output to the log
+   */
+  caller: boolean,
+  /**
+   * If this flag is enabled, $context.identity.user will be output to the log
+   */
+  user: boolean,
+  /**
+   * If this flag is enabled, $context.requestTime will be output to the log
+   */
+  requestTime: boolean,
+  /**
+   * If this flag is enabled, $context.httpMethod will be output to the log
+   */
+  httpMethod: boolean,
+  /**
+   * If this flag is enabled, $context.resourcePath will be output to the log
+   */
+  resourcePath: boolean,
+  /**
+   * If this flag is enabled, $context.status will be output to the log
+   */
+  status: boolean,
+  /**
+   * If this flag is enabled, $context.protocol will be output to the log
+   */
+  protocol: boolean,
+  /**
+   * If this flag is enabled, $context.responseLength will be output to the log
+   */
+  responseLength: boolean
+}
+
+/**
+ * factory methods for access log format.
+ */
+export class AccessLogFormat {
+  /**
+   * Generate Common Log Format.
+   */
+  public static clf(): AccessLogFormatProps {
+    return {
+      format: `${AccessLogField.contextIdentitySourceIp()} ${AccessLogField.contextIdentityCaller()} ${AccessLogField.contextIdentityUser()} \
+[${AccessLogField.contextRequestTime()}] "${AccessLogField.contextHttpMethod()} ${AccessLogField.contextResourcePath()} ${AccessLogField.contextProtocol()}" \
+${AccessLogField.contextStatus()} ${AccessLogField.contextResponseLength()} ${AccessLogField.contextRequestId()}`
+    };
+  }
+
+  /**
+   * Generate standard json log format.)
+   */
+  public static jsonWithStandardFields(
```

Comment:
Create a new test suite `access-log.test.ts` and add unit tests for this method.

1. assert all properties are part of the JSON when this method is called with no parameters.

2. assert no properties are part of the JSON when all values are set to `false`.

Reply:
Is number 1 sufficient for the following name test?
If the custom log destination log group and format is set

Reply:
Sorry, I don't understand your response. Can you re-phrase?

Reply:
I understand your comment. Add a test.I understand your comment. Add a test.